packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -159,6 +178,142 @@ async function createOutgoingPayment(
   }
 }
 
+function validateAccessLimits(
+  payment: OutgoingPayment,
+  limits: AccessLimits
+): PaymentLimits | undefined {
+  if (
+    (!limits.receiver || payment.receiver === limits?.receiver) &&
+    validateAmountAssets(payment, limits)
+  ) {
+    let paymentInterval: Interval | undefined
+    if (limits.interval)
+      paymentInterval = getInterval(limits.interval, payment.createdAt)
+    if (!limits.interval || !!paymentInterval) {
+      return { ...limits, paymentInterval }
+    }
+  }
+}
+
+function validatePaymentLimits({
+  limits,
+  payment
+}: {
+  limits: PaymentLimits
+  payment: OutgoingPayment
+}): boolean {
+  if (
+    (!limits.receiver || payment.receiver === limits?.receiver) &&
+    validateAmountAssets(payment, limits) &&
```

Comment:
I was wondering whether we have to include this check since it was already checked on payment creation of the old payments and this function is only called on old payments under the same grant.

Reply:
That relates to [this comment](https://github.com/interledger/rafiki/pull/366#discussion_r947929309)